soc/xtensa/intel_adsp/ace_v1x/include/ace-ipc-regs.h

```diff
@@ -4,15 +4,23 @@
 #ifndef ZEPHYR_SOC_INTEL_ADSP_ACE_IPC_REGS_H
 #define ZEPHYR_SOC_INTEL_ADSP_ACE_IPC_REGS_H
 
+#include <intel_adsp_ipc.h>
+
 /**
- * @remark Inter Processor Communication:
- * Used for sending interrupts to and receiving them from another
- * device. ACE uses it to talk to the host and the CSME. In general
- * there is one of these blocks instantiated for each endpoint of a
+ * @file
+ *
+ * Inter Processor Communication - used for sending interrupts to and receiving
+ * them from another device. ACE uses it to talk to the host and the CSME.
+ * In general there is one of these blocks instantiated for each endpoint of a
  * connection.
  */
-struct cavs_ipc {
 #ifdef CONFIG_SOC_SERIES_INTEL_ACE1X
+
+/**
+ * @brief IPC registers layout for Intel ADSP ACE1X SoC family.
+ */
+struct intel_adsp_ipc {
+#ifdef CONFIG_BOARD_INTEL_ADSP_ACE15_MTPM
```

Comment:
do we need the ifdef here? who else would want this? This is also limiting, and in violation of layering, BOARD definition should not be used in SOC code. Just take it out, probably you also want to remove the enclosing ifdef (`#ifdef CONFIG_SOC_SERIES_INTEL_ACE1X`) as well.

Reply:
Well those were left here on purpose... I think I can't answer your question in public without violation of current Intel trade secrets. Lets say that there are multiple flavors of MTL board in plans and ACE 1X is not a last generation of ACE family.
That is why this very well placed limiting is added - next developer adding some new code to ACE family will not use MTL regs by accident.

Reply:
> This is also limiting, and in violation of layering, BOARD definition should not be used in SOC code.

Maybe you are right, I just followed CAVS principle by analogy...

Reply:
> That is why this very well placed limiting is added - next developer adding some new code to ACE family will not use MTL regs by accident.

That should not happen, thats how we have 4 generations of CAVS now, each with their own headers and they will only be included for the right generation. Adding such conditionals is something we are trying to avoid, if we have to add them for each board/soc, that will be a real mess.